src/transaction/flashback.c

```diff
@@ -99,8 +93,10 @@ flashback_is_duplicated_request (THREAD_ENTRY * thread_p)
 	{
 	  /* - flashback_Current_conn is overwritten with new connection, so in_flashback value is initialized to false.
 	   * - previous flashback connection has been exited abnormally. */
-
-	  flashback_reset ();
+	  if (need_reset)
+	    {
+	      flashback_reset ();
```

Comment:
How about resetting in `flashback_initialize()` and leaving this function as just a checking function.